### PR TITLE
Release notes for SCP-firmware v2.12

### PR DESCRIPTION
**Pull request by AhmedImbabyGadallah**
Update the change log to summarize the changes included in this release.

### CODE REVIEW

File: change_log.md [hunk 1]
```diff
@@ -1,7 +1,84 @@
...
+    - module_common: fix FWK_SRC scoping
+
+- Platforms:
+    - synquacer: allocate secure DRAM for BL31
+
+Deprecated
+----------
+
+- Platforms:
+    - tc0: Deprecate platform
```
Comment (tarek-arm) [inline change_log.md:82]: Missing new maintainer list and scmi clock fix

**Review by tarek-arm — CHANGES_REQUESTED** [review]
Missing new maintainer list and scmi clock fix

File: change_log.md (hunk 1 above)
Comment (AhmedImbabyGadallah) [reply change_log.md:86]: done

**Review by tarek-arm — APPROVED** [review]

**Comment by nicola-mazzucato-arm** [comment]
Jenkins: ok to test

**Pull request merged**